Sources/SentryCrash/Recording/SentryCrashReport.c

```diff
@@ -1166,7 +1166,7 @@ writeBinaryImage(
     const SentryCrashReportWriter *const writer, const char *const key, const int index)
 {
     SentryCrashBinaryImage image = { 0 };
-    if (!sentrycrashdl_getBinaryImage(index, &image)) {
+    if (!sentrycrashdl_getBinaryImage(index, &image, /*isCrash*/ true)) {
```

Comment:
This is the one place where we actually need to specify to attempt grabbing the debug image's crash info.

Reply:
`h`: As this is the only place where `isCrash` is `true`, I think we don't need to bubble up `isCrash` in all the layers. The `SentryDebugImageProvider` and `PrivateSentrySDKOnly` could choose the default of `false`. The PR would be smaller and consumers of the getBinaryImage API don't have to decide if they should pass in `true` or `false` for `isCrash`. What do you think @armcknight?

Reply:
If we do that, then this becomes a breaking change not just at build time by removing the old method from `SentryDebugImageProvider`, but also at runtime because the current behavior of the public API always attempts to gather the crash info and we'd be changing that.

Another option is to decouple the image provider from collecting crash info, and extract a SentryCrashInfoProvider that would be used here on line 1172, after retrieving the debug images and before writing them. Then all the other codepaths to `sentrycrashdl_getBinaryImage` wouldn't have to specify anything, and we could even put calls to SentryCrashInfoProvider in the backwards-compat methods to maintain the current behavior.

What do you think?